iothub/service/tests/TestAssert.cs

```diff
@@ -24,20 +24,18 @@ public static class TestAssert
             }
             catch (Exception ex)
             {
-                throw new AssertFailedException(
-                    "{0}. Expected:<{1}> Actual<{2}>".FormatInvariant(errorMessage, typeof(TException).ToString(), ex.GetType().ToString()), 
-                    ex); 
+                throw new AssertFailedException($"{errorMessage}. Expected:{typeof(TException).ToString()} Actual:{ex.GetType().ToString()}", ex);
             }
 
-            throw new AssertFailedException("{0}. Expected {1} exception but no exception is thrown".FormatInvariant(errorMessage, typeof(TException).ToString()));
+            throw new AssertFailedException($"{errorMessage}. Expected {typeof(TException).ToString()} exception but no exception is thrown.");
```

Comment:
ToString()is implied by the runtime.

```suggestion
            throw new AssertFailedException($"{errorMessage}. Expected {typeof(TException)} exception but no exception is thrown.");
```